python/ml_croissant/ml_croissant/_src/operation_graph/operations/download.py

```diff
@@ -59,6 +59,27 @@ def get_download_filepath(node: Node, url: str) -> epath.Path:
     return constants.DOWNLOAD_PATH / f"croissant-{hashed_url}"
 
 
+def insert_credentials(url: str, username: str | None, password: str | None) -> str:
+    """Inserts credentials in the URL for an HTTP authentication.
+
+    Example: https://{username}:{password}@github.com/account/repo
```

Comment:
Maybe it should be

```
Returns: the URL populated with the given credentials. For example: `https://{username}...`

```

Reply:
Very good point! I fixed this in the next PR.

Reply:
(Follow-up: https://github.com/mlcommons/croissant/pull/180)